python/dgl/dataloading/dataloader.py

```diff
@@ -436,7 +436,10 @@ def sample(self, g, seed_nodes, exclude_eids=None):
             graph_device = g.device
 
         for block_id in reversed(range(self.num_layers)):
-            seed_nodes_in = to_device(seed_nodes, graph_device)
+            if not g.is_pinned():
+                seed_nodes_in = to_device(seed_nodes, graph_device)
```

Comment:
In fact I have a question here. AFAIK, `seed_nodes` here usually has been processed by `utils.prepare_tensor_or_dict()`, for example:
https://github.com/dmlc/dgl/blob/08f464348f5f4855941130356d2db9da7ba45ab9/python/dgl/dataloading/dataloader.py#L584-L586
So when would `seed_nodes` has a different context from `graph_device`? I tried commenting out this line and it threw an error.

Reply:
I see. When we might deprecate the CPU-GPU hybrid sampling in the future, we can also simplify this.

Reply:
I think we will deprecate CPU-GPU hybrid sampling as soon as this PR is merged.